example/js/api/diplomat-runtime.mjs

```diff
@@ -182,15 +182,102 @@ export class DiplomatBuf {
 	}
 
 	constructor(ptr, size, free) {
-	this.ptr = ptr;
-	this.size = size;
-	// Generated code calls one of methods these for each allocation, to either
-	// free directly after the FFI call, to leak (to create a &'static), or to
-	// register the buffer with the garbage collector (to create a &'a).
-	this.free = free;
-	this.leak = () => { };
-	this.garbageCollect = () => DiplomatBufferFinalizer.register(this, this.free);
+		this.ptr = ptr;
+		this.size = size;
+		// Generated code calls one of methods these for each allocation, to either
+		// free directly after the FFI call, to leak (to create a &'static), or to
+		// register the buffer with the garbage collector (to create a &'a).
+		this.free = free;
+		this.leak = () => { };
+		this.garbageCollect = () => DiplomatBufferFinalizer.register(this, this.free);
 	}
 }
 
+export class DiplomatWriteBuf {
+	leak;
+
+	#wasm;
+	#buffer;
+
+	constructor(wasm) {
+		this.#wasm = wasm;
+		this.#buffer = this.#wasm.diplomat_buffer_write_create(0);
+
+		this.leak = () => { };
+	}
+	
+	free() {
+		this.#wasm.diplomat_buffer_write_destroy(this.#buffer);
+	}
+
+	garbageCollect() {
+		DiplomatBufferFinalizer.register(this, this.free);
+	}
+
+	readString8() {
+		return readString8(this.#wasm, this.ptr, this.size);
+	}
+
+	get buffer() {
+		return this.#buffer;
+	}
+
+	get ptr() {
+		return this.#wasm.diplomat_buffer_write_get_bytes(this.#buffer);
+	}
+
+	get size() {
+		return this.#wasm.diplomat_buffer_write_len(this.#buffer);
+	}
+}
+
+export class DiplomatReceiveBuf {
```

Comment:
nit: document the purpose of both of these classes


For example, this appears to be solely for receiving structs, yes?

Reply:
I'll add comments right now

Reply:
Documented